pyat/at/plot/resonances.py

```diff
@@ -0,0 +1,505 @@
+"""AT plotting functions related to resonances."""
+
+import warnings
+from fractions import Fraction
+
+import matplotlib.lines as mlines
+import matplotlib.pyplot as plt
+import numpy
+from matplotlib.figure import Figure
+
+__all__ = ["farey_sequence", "plot_tune2d_resonances"]
+
+# 2024jul31 oblanco at ALBA CELLS
+
+
+def farey_sequence(nthorder: int, **kwargs: dict[str, any]) -> tuple[list, list]:
+    """
+    Return the Farey sequence, and the resonance sequence of nth order.
+
+    Parameters:
+        nthorder: natural number bigger than 0.
+        kwargs: verbose. prints extra info. Default: False.
+
+    Returns:
+        fareyseqfloat: list of elements with the Farey sequence in Float format.
+            See Eqs.(1,2,3) of [1].
+        fareyseqfrac: list of elements with the Farey sequence in Fraction format.
+            See Eqs.(1,2,3) of [1].
+
+    [1] R.Tomas. 'From Farey sequences to resonance diagrams.
+            Phys.Rev.Acc.Beams 17, 014001 (2014)'
+    """
+    # verboseprint to check flag only once
+    verbose = kwargs.pop("verbose", False)
+    verboseprint = print if verbose else lambda *a, **k: None
+    verboseprint(f"nthorder={nthorder}")
+
+    afarey = 0
+    bfarey = 1
+    cfarey = 1
+    dfarey = nthorder
+    farey = [0, 1 / dfarey]
+    fracfarey = [Fraction(0), Fraction(1, dfarey)]
+    idx = 0
+    while (farey[-1] < 1) and (idx < 100):
+        idx += 1
+        caux = numpy.floor((nthorder + bfarey) / dfarey) * cfarey - afarey
+        daux = numpy.floor((nthorder + bfarey) / dfarey) * dfarey - bfarey
+        afarey = cfarey
+        bfarey = dfarey
+        cfarey = int(caux)
+        dfarey = int(daux)
+        farey.append(cfarey / dfarey)
+        fracfarey.append(Fraction(cfarey, dfarey))
+    verboseprint(f"farey_float{nthorder}= {farey}")
+    verboseprint(f"farey_frac{nthorder} = {fracfarey}")
+    return farey, fracfarey
+
+
+def plot_tune2d_resonances(
+    **kwargs: dict[str, any],
```

Comment:
Most of the keyword arguments have an immutable default value (period, verbose, includelegend, block,…). In such a case, it's better to declare them in the function declaration, if possible with a type hint. Then they appear with their type and default value in the documentation, in the signature of the function.

Reply:
Done.